src/docfx/publish/PublishUrlMap.cs

```diff
@@ -65,6 +65,13 @@ public IEnumerable<(string url, FilePath sourcePath, MonikerList monikers)> GetP
             return _publishUrlMap.Values.SelectMany(x => x).Select(x => (x.Url, x.SourcePath, x.Monikers));
         }
 
+        public PublishUrlMapItem GeneratePublishUrlMapItem(FilePath path)
```

Comment:
It's a bit strange to expose `GeneratePublishUrlMapItem` to others.
Can we just construct the needed fields in the usage instead?

Reply:
you mean passing those `_documentProvider`, `_monikerProvider` dependencies into `PublishModelBuilder`